src/app/qgsprojectproperties.cpp

```diff
@@ -2553,8 +2553,8 @@ void QgsProjectProperties::addScaleToScaleList( QListWidgetItem *newItem )
 void QgsProjectProperties::scaleItemChanged( QListWidgetItem *changedScaleItem )
 {
   // Check if the new value is valid, restore the old value if not.
-  QRegExp regExp( "1:0*[1-9]\\d*" );
-  if ( regExp.exactMatch( changedScaleItem->text() ) )
+  QRegularExpression regExp( "1:0*[1-9]\\d*" );
```

Comment:
Needs anchoredPattern

Reply:
please check other occurrences of exactMatch have all been ported to anchoredPattern, and replace other capturedStart(...) >= 0 checks with .hasMatch().

Ping me when this is done and I'll continue the review :+1:

Reply:
Most of those should be in the other branch, as we updated dependencies, do you want me to merge the two, I was not sure how to best handle this considering the recent changes.

Reply:
let's just get this one merged and then see what changes remain in the other branch?